core/pkg/kubecost/window.go

```diff
@@ -28,8 +28,60 @@ var (
 	rfc3339             = `\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\dZ`
 	rfcRegex            = regexp.MustCompile(fmt.Sprintf(`(%s),(%s)`, rfc3339, rfc3339))
 	timestampPairRegex  = regexp.MustCompile(`^(\d+)[,|-](\d+)$`)
+
+	tOffsetLock sync.Mutex
+	tOffset     *time.Duration
+
+	utcOffsetLock sync.Mutex
+	utcOffsetDur  *time.Duration
 )
 
+// get and cache the thanos offset duration.
+// TODO: Due to dependencies here, we have to drag a non-core config option into
+// TOOD: core scope. Any solution here would be a one-off until we can generalize
+// TODO: global configuration options.
+func thanosOffset() time.Duration {
+	tOffsetLock.Lock()
+	defer tOffsetLock.Unlock()
+
+	if tOffset == nil {
+		d, err := time.ParseDuration(env.Get("THANOS_QUERY_OFFSET", "3h"))
+		if err != nil {
+			d = 0
+		}
+
+		tOffset = &d
+	}
+
+	return *tOffset
+}
+
+// returns true if thanos is enabled
+// TODO: Same note as thanosOffset above - temporary work-around until more
+// TODO: generalized global configuration.
+func isThanosEnabled() bool {
+	return env.GetBool("THANOS_ENABLED", false)
+}
+
+// returns the configured utc offset as a duration
+// TODO: Same as the above options -- we should provide a one-time initialization configuration
+// TODO: for these values, or deprecate their use.
+func utcOffset() time.Duration {
```

Comment:
This is another case where we've leaked opencost env configuration into our utilities. I think this can be solved next pass (it's also possible that UTC Offset configuration is no longer a valid config option).